tensorflow/compiler/xla/service/gpu/horizontal_fusion_test.cc

```diff
@@ -0,0 +1,367 @@
+}
+
+TEST_F(HorizontalFusionTest, RMSPropLike) {
+  HloComputation::Builder builder(TestName());
+
+  std::vector<HloInstruction*> all_outputs;
+  for (size_t i = 0; i < 48; ++i) {
+    auto shape = ShapeUtil::MakeShape(F32, {2, 1024 + i});
+    // ms <- grad**2 (1 - rho) + ms * rho
+    HloInstruction* grad = builder.AddInstruction(
+        HloInstruction::CreateParameter(i * 9 + 0, shape, "grad"));
+    HloInstruction* ms = builder.AddInstruction(
+        HloInstruction::CreateParameter(i * 9 + 1, shape, "ms"));
+    HloInstruction* rho =
+        builder.AddInstruction(HloInstruction::CreateParameter(
+            i * 9 + 2, ShapeUtil::MakeShape(F32, {}), "rho"));
+    HloInstruction* one_minus_rho =
+        builder.AddInstruction(HloInstruction::CreateParameter(
+            i * 9 + 3, ShapeUtil::MakeShape(F32, {}), "one_minus_rho"));
+    auto rho_broadcasted =
+        builder.AddInstruction(HloInstruction::CreateBroadcast(shape, rho, {}));
+    auto one_mins_rho_broadcasted = builder.AddInstruction(
+        HloInstruction::CreateBroadcast(shape, one_minus_rho, {}));
+    auto grad_squared = builder.AddInstruction(
+        HloInstruction::CreateBinary(shape, HloOpcode::kMultiply, grad, grad));
+    auto ms_1st_term = builder.AddInstruction(HloInstruction::CreateBinary(
+        shape, HloOpcode::kMultiply, grad_squared, one_mins_rho_broadcasted));
+    auto ms_2nd_term = builder.AddInstruction(HloInstruction::CreateBinary(
+        shape, HloOpcode::kMultiply, ms, rho_broadcasted));
+    auto ms_out = builder.AddInstruction(HloInstruction::CreateBinary(
+        shape, HloOpcode::kAdd, ms_1st_term, ms_2nd_term));
+
+    // mom <- momentum * mom_{t-1} + lr * grad / sqrt(ms + epsilon)
+    HloInstruction* momentum = builder.AddInstruction(
+        HloInstruction::CreateParameter(i * 9 + 4, shape, "momemtum"));
+    HloInstruction* mom = builder.AddInstruction(
+        HloInstruction::CreateParameter(i * 9 + 5, shape, "mom"));
+    HloInstruction* lr = builder.AddInstruction(HloInstruction::CreateParameter(
+        i * 9 + 6, ShapeUtil::MakeShape(F32, {}), "lr"));
+    HloInstruction* epsilon =
+        builder.AddInstruction(HloInstruction::CreateParameter(
+            i * 9 + 7, ShapeUtil::MakeShape(F32, {}), "epsilon"));
+    auto lr_broadcasted =
+        builder.AddInstruction(HloInstruction::CreateBroadcast(shape, lr, {}));
+    auto epsilon_broadcasted = builder.AddInstruction(
+        HloInstruction::CreateBroadcast(shape, epsilon, {}));
+    auto mom_1st_term = builder.AddInstruction(HloInstruction::CreateBinary(
+        shape, HloOpcode::kMultiply, momentum, mom));
+    auto ms_eps = builder.AddInstruction(HloInstruction::CreateBinary(
+        shape, HloOpcode::kAdd, ms_out, epsilon_broadcasted));
+    auto ms_eps_rsq = builder.AddInstruction(
+        HloInstruction::CreateUnary(shape, HloOpcode::kRsqrt, ms_eps));
+    auto grad_ms_eps_rsq = builder.AddInstruction(HloInstruction::CreateBinary(
```

Comment:
`grad_ms_eps_rsq` is unused. Can the line be removed or just the local variable?

Reply:
Fixed. The wiring was not as intended. My oversight.